src/QMCDrivers/QMCDriverNew.cpp

```diff
@@ -287,16 +288,10 @@ void QMCDriverNew::createRngsStepContexts(int num_crowds)
   step_contexts_.resize(num_crowds);
   Rng.resize(num_crowds);
 
-  if (RandomNumberControl::Children.size() == 0)
-  {
-    app_warning() << "  Initializing global RandomNumberControl! "
-                  << "This message should not be seen in production code but only in unit tests." << std::endl;
-    RandomNumberControl::make_seeds();
-  }
-
+  auto& rng_children = RandomNumberControl::getChildren();
```

Comment:
Complete havoc if in testing I create two QMCDriverNew derived classes, the sematics of this are bizarre and exist because the optimizer was not really improved for batched. QMCDriver should take some sort of random context in to retain determinism but which does not bleed how many crowds/threads there are into the global scope. The function should return some updated RandomContext perhaps with just its starting offset updated or whatever. If test code or the main was spawning another driver it could then also update the and no dependence on this global object with global OMP thead semantics to some degree assumed. Also just no invisible single rng collection state machine. 

This could also solve the very annoying problem in tests where multiple  tests cases that use RNG can end up broken when different subcollections of tests are ran.  A useful capability of catch which this global mess completely destroys.

Reply:
I noticed the complaint in the source code comment and it is time to change.